frontend-ssr/pages/submissions.tsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import Head from 'next/head';
+import Footer from '~/components/footer/Footer.component';
+import Header from '~/components/header/Header.component';
+
+import styles from '../styles/pages/default-page.module.scss';
+import style from './submissions.module.scss';
+import ExSubmission from '../components/exercise-submissions/ExerciseSubmission';
+import ExerciseService from '../services/ExerciseSubmissions.service';
+import { loadSubmissions } from '~/redux/exercise-submissions/exercise-submissions.actions';
+import SubmissionSkeleton from '~/components/exercise-submissions/ExerciseSubmissionSkeleton';
+import SweetAlertService from '../services/sweet-alert/SweetAlert.service';
+import store from '~/redux/store';
+
+interface State {
+  loading: boolean;
+}
+interface Props {
+  submissions: any;
+  dispatch: any;
+}
+
+class Submissions extends React.Component<Props, State> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      loading: true,
+    };
+  }
+
+  componentDidMount() {
+    let { submissions } = this.props;
+    const state = submissions;
+    function shouldFetchSubmissions() {
+      if (state.submissions === undefined) {
+        return true;
+      }
+      return false;
+    }
+    if (shouldFetchSubmissions()) {
+      this.getData();
+    } else {
+      store.getState();
+      this.setState({
+        loading: false,
+      });
+    }
+  }
+
+  getData = async () => {
+    const { dispatch } = this.props;
+    try {
+      await ExerciseService.getSubmissions(0, '')
+        .then((resp) => resp.json())
+        .then((jsonResp) => dispatch(loadSubmissions(jsonResp)));
+      this.setState({
+        loading: false,
+      });
+    } catch (err) {
+      SweetAlertService.toast({ type: 'error', text: err });
+    }
+  };
+
+  render() {
+    const { loading } = this.state;
+    const { submissions } = this.props;
+    return (
+      <>
+        <Head>
+          <title>HTML</title>
+          <link rel="icon" href="/favicon.ico" />
+        </Head>
+        <Header />
+        <main className={styles.page}>
+          <div className={`${style['page-submissions']}`}>
+            <ul className={style.cardWrapper}>
+              {submissions.submissions !== undefined
+                ? submissions.submissions.map((submission) => (loading
```

Comment:
This doesn't work, I never see the skeletons even if I specifically delay the response.

There's a big flaw in this code, I'm gonna let you look for it first and if you don't find it I'll point you in the right direction :D